src/modules/txbuilder/RadixTransactionBuilder.ts

```diff
@@ -829,30 +822,63 @@ export default class RadixTransactionBuilder {
         return atom
     }
 
+    private static assertMinNativeTokenBalance(account: RadixAccount, isAtLeast: BN) {
+        const requiredBalance = isAtLeast
+        const nativeTokenBalance = account.transferSystem.snapshotOfNativeTokenBalance()
+
+        if (nativeTokenBalance.lt(requiredBalance)) {
+            throw new Error(`${account.address.toString()} owns
+             only ${RadixTokenDefinition.fromSubunitsToDecimal(nativeTokenBalance)} 
+             XRDS but expected at least 
+             ${RadixTokenDefinition.fromSubunitsToDecimal(requiredBalance)}
+             `)
+        }
+        // ok!
+    }
+
+    public static addFeeToAtom(atom: RadixAtom, account: RadixAccount) {
```

Comment:
same as above

Reply:
can't since `particleGroups` in TXBuilder is `private`...